api/v1beta1/nutanixmachine_types.go

```diff
@@ -70,8 +93,13 @@ type NutanixMachineSpec struct {
 	// image is to identify the rhcos image uploaded to the Prism Central (PC)
 	// The image identifier (uuid or name) can be obtained from the Prism Central console
 	// or using the prism_central API.
-	// +kubebuilder:validation:Required
-	Image NutanixResourceIdentifier `json:"image"`
+	// +kubebuilder:validation:Optional
+	// +optional
+	Image *NutanixResourceIdentifier `json:"image,omitempty"`
```

Comment:
we are going from required to now both optional. how are we conveying to user if they dont given any of it. can we  do something like following?
<pre>
image: <=== mandatory
      id *NutanixResourceIdentifier <==optional if imageLookup is provided
      imageLookup: <=== optional if id is provided
      ...
</pre>

Reply:
https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/498/files#diff-22f4b13037c2fbd14d0c08f995ed39a4460b298da33bc3d4b1d718f25da0f14bR73

this is handled by the CEL expression adding the line below in case that link doesn't work

```
// +kubebuilder:validation:XValidation:rule="has(self.image) != has(self.imageLookup)",message="Either 'image' or 'imageLookup' must be set, but not both"
``` 

if they are both missing it will be `false != false` which will fail.